src/shave.js

```diff
@@ -14,6 +14,11 @@ export default function shave (target, maxHeight, opts = {}) {
     const styles = el.style
     const span = el.querySelector(`.${classname}`)
     const textProp = el.textContent === undefined ? 'innerText' : 'textContent'
+    const parsedMaxHeight = (typeof maxHeight === 'number')
```

Comment:
My issue is that if the `maxHeight` is not a number and doesn't have a percentage (`%`), then what is the `maxHeight`. Should it be an error? I'm not sure. What are your thoughts?

Reply:
As of now, I just decided to return the same parameter that was being parsed, so that the previous implementation would execute (basically, parsing the parameter would do nothing).

But since it's not a number (which the previous implementation seems to need), the best thing to do should be to throw an error.

Reply:
Alright, could you make that update? ~THANK YOU!

Reply:
Sure, I also found something strange while using shave, which I'm now working on.

Basically, the height comparison happens between the maxHeight argument, and the element height that is calculated based on the text (which is calculated automatically based on the environment around the text, css width and some other stuff, when you replace the height style to auto). 

The problem is that I was seeing some text without the ellipsis on the page I'm working on, and since I was using the modified shave with percentage parsing, and I was using 100% of the height, i started suspecting something was off with the way we were detecting the original full height of the visible space that the text can be displayed on.

After trying some stuff, I found out that using **clientHeight** instead of **offsetHeight** fixed the problem somehow.

Is there any reason we are using **offsetHeight** instead of **clientHeight**?

Reply:
I'll open some other PRs for other modifications I needed for this project, to avoid bloating this PR.

Reply:
As the scope of the feature grows, the amount of potential new issues increases. 

The calculation that is done to provide this feature within `shave` could be done could be provided into the `shave`'s `maxHeight` argument so this feature is not needed to accomplish the same result.  

@juanmardefago, I request this update be a **readme** example only which shows how to calculate the `maxHeight` from a percentage and pass it in as `shave`'s `maxHeight` argument. 

Thank you! Sorry for using any of your extra time.

Reply:
Sure, I'll modify the PR so I only add an explanation on how to manually calculate the maxHeight on the README.md.

I'll keep these parsing changes (with the corresponding fixes to any issues I find) in my own fork of shave, and once everything is fixed there I'll do some performance testing.

If you want I can keep you updated with both the results and any impact it has, in case you might want to include it to the main branch sometime down the road ;)